analysis/import/SonarImporter/src/main/kotlin/de/maibornwolff/codecharta/importer/sonar/SonarImporterMain.kt

```diff
@@ -96,15 +96,23 @@ class SonarImporterMain(
 
     override fun getDialog(): ParserDialogInterface = ParserDialog
     override fun isApplicable(resourceToBeParsed: String): Boolean {
-        val trimmedInput = resourceToBeParsed.trim()
+        println("Checking if SonarImporter is applicable...")
+        val searchFile = "sonar-project.properties"
 
-        if (trimmedInput.contains("^http(s)?://".toRegex())) {
+        val trimmedInput = resourceToBeParsed.trim()
+        val inputFile = File(trimmedInput)
+        if (trimmedInput.contains("^http(s)?://".toRegex()) || (inputFile.isFile && inputFile.name == searchFile)) {
             return true
         }
+        if (!inputFile.isDirectory || trimmedInput == "") {
+            return false
+        }
```

Comment:
I would recommend to combine these if-statements.
One Idea would be to have a private method for `(inputFile.isFile && inputFile.name == searchFile)` like `isSonarPropertiesFile(...)`. That would make it more readable.
Another Example: `isUrl()`